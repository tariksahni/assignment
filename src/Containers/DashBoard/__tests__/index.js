import React from 'react'
import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import { Provider } from 'react-redux'
import { shallow } from 'enzyme'
import { withRouter } from 'react-router'
import renderer from 'react-test-renderer'
import configureStore from '../../../Store'
import { DashBoard } from '../../'

const store = configureStore()

configure({ adapter: new Adapter() })

describe(DashBoard, () => {
  const Component = shallow(<Provider store={store}><DashBoard /></Provider>)
  const WrappedComponent = withRouter(Component)
  it('renders and matches our snapshot', () => {
    const Component = <Provider store={store}><DashBoard /></Provider>
    const WrappedComponent = withRouter(Component)
    const component = renderer.create(WrappedComponent)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('contains the supplied name', () => {
    expect(WrappedComponent.text()).toContain('<Connect(DashBoard) />')
  })
})
