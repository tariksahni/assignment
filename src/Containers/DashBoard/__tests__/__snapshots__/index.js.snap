// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`function Connect(props, context) {
        _classCallCheck(this, Connect);

        var _this = _possibleConstructorReturn(this, _Component.call(this, props, context));

        _this.version = version;
        _this.state = {};
        _this.renderCount = 0;
        _this.store = props[storeKey] || context[storeKey];
        _this.propsMode = Boolean(props[storeKey]);
        _this.setWrappedInstance = _this.setWrappedInstance.bind(_this);

        (0, _invariant2.default)(_this.store, 'Could not find "' + storeKey + '" in either the context or props of ' + ('"' + displayName + '". Either wrap the root component in a <Provider>, ') + ('or explicitly pass "' + storeKey + '" as a prop to "' + displayName + '".'));

        _this.initSelector();
        _this.initSubscription();
        return _this;
      } renders and matches our snapshot 1`] = `null`;
